Source/ProgressionSystemRuntime/Private/Components/PSSpotComponent.cpp

```diff
@@ -50,13 +61,55 @@ void UPSSpotComponent::OnUnregister()
 	Super::OnUnregister();
 }
 
+// Called when the end game state was changed to toggle progression widget visibility.
+void UPSSpotComponent::OnGameStateChanged_Implementation(ECurrentGameState CurrentGameState)
+{
+	APlayerCharacter* PlayerCharacter = UMyBlueprintFunctionLibrary::GetLocalPlayerCharacter();
+	if (!ensureMsgf(PlayerCharacter, TEXT("ASSERT: [%i] %s:\n'PlayerCharacter' is not valid!"), __LINE__, *FString(__FUNCTION__)))
```

Comment:
Per my previous review, please dont use *FString(__FUNCTION__) anymore, but __FUNCTION__ along with %hs for any new code

Reply:
done https://github.com/moinm3uw/ProgressionSystem/pull/59/commits/4b48f10c8cf09fe5972ce1e7cc9f512936f18b9b